src/storage/src/render/multi_worker_persist_sink.rs

```diff
@@ -0,0 +1,1123 @@
+        let mut desired_frontier;
+
+        loop {
+            if let Some(event) = desired_input.next().await {
+                match event {
+                    Event::Data(_cap, _data) => {
+                        // Just read away data.
+                        // TODO(guswynn): this, and the same code in the compute version of this
+                        // code, is inefficient, and can be improved.
+                        // See
+                        // <https://github.com/MaterializeInc/materialize/pull/17589#discussion_r1106016139>
+                        // for more info.
+                        continue;
+                    }
+                    Event::Progress(frontier) => {
+                        desired_frontier = frontier;
+                    }
+                }
+            } else {
+                // Input is exhausted, so we can shut down.
+                return;
+            };
+
+            // If the new frontier for the data input has progressed, produce a batch description.
+            if PartialOrder::less_than(&current_upper, &desired_frontier) {
+                // The maximal description range we can produce.
+                let batch_description = (current_upper.to_owned(), desired_frontier.to_owned());
+
+                let lower = batch_description.0.first().unwrap();
+                let batch_ts = batch_description.0.first().unwrap().clone();
+
+                let cap = cap_set
+                    .try_delayed(&batch_ts)
+                    .ok_or_else(|| {
+                        format!(
+                            "minter cannot delay {:?} to {:?}. \
+                                Likely because we already emitted a \
+                                batch description and delayed.",
+                            cap_set, lower
+                        )
+                    })
+                    .unwrap();
+
+                trace!(
+                    "persist_sink {collection_id}/{shard_id}: \
+                        new batch_description: {:?}",
+                    batch_description
+                );
+
+                output.give(&cap, batch_description).await;
+
+                // WIP: We downgrade our capability so that downstream
+                // operators (writer and appender) can know when all the
+                // writers have had a chance to write updates to persist for
+                // a given batch. Just stepping forward feels a bit icky,
+                // though.
+                let new_batch_frontier = Antichain::from_elem(batch_ts.step_forward());
+                trace!(
+                    "persist_sink {collection_id}/{shard_id}: \
+                        downgrading to {:?}",
+                    new_batch_frontier
+                );
+                let res = cap_set.try_downgrade(new_batch_frontier.iter());
+                match res {
+                    Ok(_) => (),
+                    Err(e) => panic!("in minter: {:?}", e),
```

Comment:
We should just call `cap_set.downgrade()` if we don't have any additional context to add to the panic